recipes/client.rb

```diff
@@ -86,38 +86,40 @@
         end
       elsif(item['enabled'] == true)
         if(item['pub_key'])
-          key_file = "#{Chef::Config[:file_cache_path]}/#{item['client']}.pub"
-          file key_file do
-            backup false
-            content item['pub_key']
-            mode '0400'
-          end
-        end
-        item['full_name'] = item.fetch('full_name', item['client'].capitalize)
-        first_name = item['full_name'].split(' ').first.capitalize
-        last_name = item['full_name'].split(' ').last.capitalize
-        email = item.fetch('email', "#{item['client']}@example.com")
-        execute "create user: #{item['client']}" do
-          command "chef-server-ctl user-create #{item['client']} #{first_name} #{last_name} #{email} #{item['password']} > /dev/null 2>&1"
-          not_if "chef-server-ctl user-list | grep '^#{item['client']}$'"
-        end
-        if(item['pub_key'])
-          execute "set user key: #{item['client']}" do
-            if node['chef-server'][:version].to_f >= 12.1
-              command "chef-server-ctl add-user-key #{item['client']} --public-key-path #{key_file} --key-name populator"
-            else
-              command "chef-server-ctl add-user-key #{item['client']} #{key_file} --key-name populator"
+          unless item['pub_key'].empty?
```

Comment:
Man, I clearly was staring at this too long b/c I spent about 20 minutes trying to think of how to handle non-strings :/ 👍